test/integration/targets/module_utils/library/test_heuristic_log_sanitize.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils import basic
+from ansible.module_utils.basic import AnsibleModule
+
+heuristic_log_sanitize = basic.heuristic_log_sanitize
+
+
+def heuristic_log_sanitize_spy(*args, **kwargs):
+    heuristic_log_sanitize_spy.return_value = heuristic_log_sanitize(*args, **kwargs)
+    return heuristic_log_sanitize_spy.return_value
+
+
+basic.heuristic_log_sanitize = heuristic_log_sanitize_spy
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec={
+            'data': {
+                'type': 'str',
+                'required': True,
+            }
+        },
+    )
+
+    # This test module is testing that the data that will be used for logging
+    # to syslog properly sanitizes URLs that contain a password.
+    #
+    # As such, we build an expected sanitized string from the input, to
+    # compare it with the output from heuristic_log_sanitize.
+    #
+    # To test this in the same way that modules ultimately operate this test
+    # monkeypatches ansible.module_utils.basic to store the sanitized data
+    # for later inspection.
+    data = module.params['data']
```

Comment:
@mattclay what do you think about the above comment explaining the test.

FYI there are some units for this, but they don't directly add to the incidental test coverage.